src/stores/federation.js

```diff
@@ -0,0 +1,181 @@
+/**
+ * @copyright Copyright (c) 2024 Maksim Sukharev <antreesy.web@gmail.com>
+ *
+ * @author Maksim Sukharev <antreesy.web@gmail.com>
+ *
+ * @license AGPL-3.0-or-later
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+import { defineStore } from 'pinia'
+import Vue from 'vue'
+
+import { FEDERATION } from '../constants.js'
+import { getShares, acceptShare, rejectShare } from '../services/federationService.js'
+
+/**
+ * @typedef {object} Share
+ * @property {string} accessToken the invitation access token
+ * @property {number} id the invitation id
+ * @property {number} localRoomId the invitation local room id
+ * @property {number} remoteAttendeeId the invitation remote attendee id
+ * @property {string} remoteServerUrl the invitation remote server URL
+ * @property {string} remoteToken the invitation remote token
+ * @property {string} roomName the invitation room name
+ * @property {number} state the invitation state
+ * @property {string} userId the invitation user id
+ * @property {string} inviterCloudId the inviter cloud id
+ * @property {string} inviterDisplayName the inviter display name
+ */
+
+/**
+ * @typedef {object} State
+ * @property {{[key: string]: Share}} pendingShares - pending invitations
+ * @property {{[key: string]: Share}} acceptedShares - accepted invitations
+ */
+
+/**
+ * Store for other app integrations (additional actions for messages, participants, e.t.c)
+ *
+ * @param {string} id store name
+ * @param {State} options.state store state structure
+ */
+export const useFederationStore = defineStore('federation', {
+	state: () => ({
+		pendingShares: {},
+		acceptedShares: {},
+	}),
+
+	actions: {
+		/**
+		 * Fetch pending invitations and keep them in store
+		 *
+		 */
+		async getShares() {
+			try {
+				const response = await getShares()
+				response.data.ocs.data
+					.forEach(item => {
```

Comment:
It's an object not iterable

Reply:
I see an array here: 
![image](https://github.com/nextcloud/spreed/assets/93392545/e7c415dd-2691-4d90-aada-1b91f9013fa6)

Reply:
Strange, sometimes it throws an error that response.data.ocs.data is not an array.